Utils/Dataflow/095_datasetInfoAMI/amiDatasets.py

```diff
@@ -81,18 +91,42 @@ def init_ami_client(userkey='', usercert=''):
                          "proxy.\n")
         raise DataflowException("Failed to initialise AMI client: certificate "
                                 "not provided or not found.")
+    try:
+        ami_client.execute('ListCommands')
+    except Exception as e:
+        sys.stderr.write("(WARN) Failed to perform test command "
+                         "ListCommands. Exception: %s\n" % str(e))
```

Comment:
I have a feeling that we have discussed this question before, which is captured in the commit log of 05283d63979d6635e16ebf75cabd58d90931ecb7:
```
Information about some datasets is available from AMI without authentication -
display warning instead of stopping the stage if the test command fails, to
allow such datasets to be processed.
```
, so I'm sorry for not being self-consistent... but now here's what I think.

This function (`init_ami_client()`) raises `DataflowException` whenever there's some sign of a failure to establish the connection (https://github.com/PanDAWMS/dkb/pull/282/files#diff-02fcd172861b2b59cff44046a135be30R79-R93):
* no `pyAMI` module (pretty understood);
* any `Exception` thrown by `pyAMI` (although we wil get here only if... I don't know, pass non-string values as `{key|cert}_file`);
* no certificate provided (although `pyAMI` would fall back to password authentication, still providing info for those datasets that do not require special authorization).

First two cases are understood: we can not init the client at all. But the third is much the same as the case of invalid certificate; so why would we fail in one case but try to proceed in another?

Reply:
If I understand this correctly, no we are changing our mind and aborting the procession if something is wrong with credentials. This is fine by me, made the changes.